Core/Src/main.c

```diff
@@ -143,7 +172,34 @@ void SystemClock_Config(void)
 }
 
 /* USER CODE BEGIN 4 */
+uint16_t ADC_handle(void){
+
+	//Pull Chip select pin low to start connection
+	HAL_GPIO_WritePin(GPIOB, CS_PIN, 0);
+	uint8_t startByte = 0x01;
+
+	//send start bit
+	uint8_t garbage;
+	HAL_SPI_TransmitReceive(&hspi1, &startByte, &garbage, 1, 100);
+
+
+	uint8_t send[2], receive[2];
+	send[0] = (1<<7);
 
+	//Set channel to 0 and receive result of conversion
+	HAL_SPI_TransmitReceive(&hspi1, send, receive, 2, 100);
+
+	//Mask extra bits
+	receive[0] &= 0x03;
+
+	//Join 10 bit integer result into 16bit int
+	uint16_t result = (receive[0]<<8)| receive[1];
```

Comment:
This looks good based on how you implemented the transmit-receive, but if you change it to an array of 3, make sure to adjust this accordingly.